File: base-class-module/src/main/java/jingya/com/base_class_module/ActivityController.java
package jingya.com.base_class_module;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

/**
 * @author kuky
 * @description Activity 管理类
 */
public class ActivityController {
    private static List<Activity> activities = new ArrayList<>();

    /**
     * onCreate 调用
     *
     * @param activity
     */
    public static void addActivity(Activity activity) {
        activities.add(activity);
    }

    /**
     * onDestroy 调用
     *
     * @param activity
     */
    public static void removeActivity(Activity activity) {
        activities.remove(activity);
        activity.finish();
    }

    /**
     * 获取栈顶 Activity
     *
     * @return
     */
    public static Activity getTopActivity() {
        if (!activities.isEmpty()) {
            return activities.get(activities.size() - 1);
        } else {
            return null;
        }
    }

    /**
     * 关闭所有 Activity
     */
    public static void finishAll() {
        if (activities.isEmpty()) {
            return;
        }

        for (Activity activity : activities) {
            if (!activity.isFinishing()) {
                activity.finish();
            }
        }
    }
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/ActivityController.kt
package com.kuky.base_kt_module

import android.app.Activity

/**
 * @author kuky.
 * @description
 */
object ActivityController {

    private val activities = ArrayList<Activity>()

    /**
     * onCreate 调用
     */
    fun addActivity(activity: Activity) {
        activities.add(activity)
    }

    /**
     * onDestroy 调用
     */
    fun removeActivity(activity: Activity) {
        if (activities.contains(activity)) {
            activities.remove(activity)
            activity.finish()
        }
    }

    /**
     * 获取栈顶 Activity
     */
    fun getTopActivity(): Activity? = if (activities.isEmpty()) null else activities[activities.size - 1]

    /**
     * 关闭所有 Activity
     */
    fun finishAll() {
        for (a in activities)
            if (!a.isFinishing) a.finish()
    }
}